docs/source/udf.rst

```diff
@@ -64,16 +66,30 @@ to a single value:
    :code:
 
 
-Here is another example, demonstrating `kind="sig"` buffers and the merge function:
+Here is another example, demonstrating :code:`kind="sig"` buffers and the merge function:
 
 
 .. include:: udf/max.py
    :code:
 
 
-For a more complete example, please have a look at the functions implemented in `libertem.udf`,
-for example `blobfinder`.
+For a more complete example, please have a look at the functions implemented in the sub-modules of :code:`libertem.udf`,
+for example :code:`libertem.udf.blobfinder`.
+
+Auto UDF
+--------
 
+The :code:`AutoUDF` class and :code:`run_auto()` function allow to run simple functions that accept a frame as the only parameter with an auto-generated :code:`kind="nav"` result buffer over a dataset ad-hoc without defining an UDF class. The :code:`AutoUDF` class determines the output shape and type by calling the function with a mock-up frame of the same type and shape as a real detector frame and converting the return value to a numpy array. The :code:`extra_shape` and :code:`dtype` parameters for the result buffer are derived automatically from this numpy array. Additional constant parameters can be passed to the function via :code:`functools.partial`, for example.
+
+Example: Calculate sum over the last signal axis.
+
+.. code-block:: python
+
+      result = libertem.udf.run_auto(
```

Comment:
Hmm, looking at this example, maybe we could make it even easier:

```
ctx.map(dataset=dataset, f=functools.partial(np.sum, axis=-1), roi=roi)
```

As this is more of a 'facility' than a concrete UDF, I would argue it has its place in the `Context`. 

We should also have a general discussion about the `Context` API, what to add vs. what not to add there etc. - this also comes into play when thinking about breaking LiberTEM into more coarse modules that are separately useful. Maybe something for Monday? cc @kruzaeva